MSI/Day10/다트게임.js

```diff
@@ -0,0 +1,35 @@
+function solution(dartResult) {
+    let score = 0;
+    const answer = [];
+    let temp = 0;
+
+    for (let i = 0; i < dartResult.length; i++) {
+        if (dartResult[i] >= 0 && dartResult[i] <= 9) {
+            if (dartResult[i] == 1 && dartResult[i + 1] == 0) {
+                temp = 10;
+                i++;
+            } else {
+                temp = dartResult[i];
+            }
+        } else if (dartResult[i] === "S") {
+            answer.push(temp);
+        } else if (dartResult[i] === "D") {
+            answer.push(Math.pow(temp, 2));
+        } else if (dartResult[i] === "T") {
+            answer.push(Math.pow(temp, 3));
+        } else if (dartResult[i] === "#") {
+            answer[answer.length - 1] *= -1;
+        } else if (dartResult[i] === "*") {
+            answer[answer.length - 1] *= 2;
+            if (answer.length > 1) {
+                answer[answer.length - 2] *= 2;
+            }
+        }
+    }
```

Comment:
```suggestion
for (let i = 0; i < dartResult.length; i++) {
    if (dartResult[i] >= 0 && dartResult[i] <= 9) {
        if (dartResult[i] == 1 && dartResult[i + 1] == 0) {
            temp = 10;
            i++;
        } else {
            temp = dartResult[i];
        }
    } else {
        switch (dartResult[i]) {
            case "S":
                answer.push(temp);
                break;
            case "D":
                answer.push(Math.pow(temp, 2));
                break;
            case "T":
                answer.push(Math.pow(temp, 3));
                break;
            case "#":
                answer[answer.length - 1] *= -1;
                break;
            case "*":
                answer[answer.length - 1] *= 2;
                if (answer.length > 1) {
                    answer[answer.length - 2] *= 2;
                }
                break;
        }
    }
}
```
조건이 많고  분기처리를 해야할때는 `switch`문을 사용해보는건 어떨까요? `else-if`가 너무 많아서 개인적으로는 가독성이 조금 떨어지는 것 같아요🙄

Reply:
확실히 깔끔해졌어요 수정했습니다 감사합니다!